### PR TITLE
Replace Travis CI with GitHub actions

### LINKED ISSUE #589: Switch to GitHub actions
GitHub actions is faster, easier and better to use.
IMO, I think it would be nice to change to GitHub actions.

### PR DESCRIPTION
**Pull request by Panquesito7**
Closes #589.

### CODE REVIEW

**Comment by ClobberXD** [comment]
Seems to be working alright in Panquesito7/capturetheflag: https://github.com/Panquesito7/capturetheflag/actions/runs/88556647

Will review this a little later. Thanks for the PR.

**Comment by ClobberXD** [comment]
I'll leave the reviewing to @rubenwardy who's more knowledgeable than me in this area.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: build
+on: [push, pull_request]
+
+jobs:
+  luacheck:
+    runs-on: ubuntu-latest
```
Comment (rubenwardy) [inline .github/workflows/build.yml:6]: is this docker-based? `abaez/luarocks` is an image with luarocks

Comment (Panquesito7) [reply .github/workflows/build.yml:6]: Yes, the build uses `docker`.

Comment (Panquesito7) [reply .github/workflows/build.yml:6]: @rubenwardy we can use [Factorio LuaCheck](https://github.com/Roang-zero1/factorio-mod-luacheck) and use local `.luacheckrc` config. It is docker-based.

**Comment by Panquesito7** [comment]
@rubenwardy @LoneWolfHT ping?

**Pull request merged**